h/tasks/indexer.py

```diff
@@ -55,12 +58,21 @@ def sync_annotations(limit):
 def report_job_queue_metrics():
     queue = celery.request.find_service(name="search_index")._queue
 
+    now = datetime.utcnow()
+
     newrelic.agent.record_custom_metrics(
         [
+            (
+                "Custom/Job/Queue/Length",
+                celery.request.db.query(Job).count(),
+            ),
+            (
+                "Custom/Job/Queue/Expired/Length",
+                celery.request.db.query(Job).filter(Job.expires_at < now).count(),
+            ),
```

Comment:
Like the unfiltered job queue length (`"Custom/Job/Queue/Length"`) I think the count of expired jobs should be just a general count and not specific to `sync_annotation` jobs.

The general idea with expired jobs is that jobs should never expire. If they do expire jobs don't get deleted but they get ignored by the queue consumer (at least, that's how the `sync_annotation` consumer works) so that stuck jobs don't pile up and block other jobs. And I'm planning to add a New Relic alarm that triggers if there are _any_ expired jobs, as this should never happen.

If we add new types of job in the future my guess is that they'll follow this same behaviour as the `sync_annotation` jobs do for expired jobs. I _don't_ imagine we'll be adding new types of jobs that we expect to expire, and not wanting them to show in the expired jobs count in New Relic or trigger the expired jobs alarm.

So I've made the expired jobs metric a general queue metric and not a sync annotations queue metric.

Reply:
Not super thrilled about the DB stuff leaking out again, but it's metrics, at this point who cares?